### PR TITLE
Add a job for 32 bit Linux

### PR DESCRIPTION
**Pull request by mrkkrp**
Ticket: https://ghc.haskell.org/trac/ghc/ticket/14598.
Patch (should be discarded if we merge this): https://phabricator.haskell.org/D4425.

It's best to merge #104 first, because:

1. The job required a custom docker image, because the existing image `haskell:8.2` did not work (`./configure` failed, and even if I could omit some options and make it work then `make` failed in all cases), so I had to create something similar based on `i368/debian:jessie` which worked. And while I was at it, I decided to make it properly with a non-root user.

2. Thus `system-prepare.sh` also should use `sudo` and so it makes sense to start this branch from `do-not-run-on-circle-ci-as-root`.

If there is a good reason, I could rebase against master and make it work without #104 somehow.

----

So this builds but a good number of tests fails (at least locally, still waiting for it to finish on CI). Fixing all those tests is for another ticket, probably.

```
SUMMARY for test run started at Mon Mar  5 13:39:08 2018 UTC
 0:44:01 spent to go through
    6273 total tests, which gave rise to
   24698 test cases, of which
   18122 were skipped

      56 had missing libraries
    6202 expected passes
      94 expected failures

       2 caused framework failures
      37 caused framework warnings
       0 unexpected passes
     222 unexpected failures
       0 unexpected stat failures
```

### CODE REVIEW

**Comment by mrkkrp** [comment]
OK, results are a bit better when this is run on CI:

```
UMMARY for test run started at Mon Mar  5 15:16:37 2018 UTC
 0:18:27 spent to go through
    6262 total tests, which gave rise to
   24654 test cases, of which
   18092 were skipped

      56 had missing libraries
    6404 expected passes
      94 expected failures

       0 caused framework failures
      36 caused framework warnings
       0 unexpected passes
       8 unexpected failures
       0 unexpected stat failures
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -226,16 +212,35 @@ jobs:
...
+    # - validate-x86_64-linux
     # FreeBSD disabled: https://github.com/haskell/unix/issues/102
     # - validate-x86_64-freebsd
-    - validate-x86_64-darwin
-    - validate-x86_64-linux-llvm
-    - validate-hadrian-x86_64-linux
+    # - validate-x86_64-darwin
+    # - validate-x86_64-linux-llvm
+    - validate-i386-linux
+    # - validate-hadrian-x86_64-linux
```
Comment (bgamari) [inline .circleci/config.yml]: I would be happy to merge this and fix up the issues in a later PR if this were removed.

**Comment by mrkkrp** [comment]
@bgamari Right, that was just a way to make test runs faster, fixed.

**Comment by bgamari** [comment]
Awesome. Let's merge this.

We also ought to move the Docker images back to the GHC organization (and perhaps give you access).

**Comment by mrkkrp** [comment]
I can push them under the ghcci account on docker hub if you give me access and then open a PR to update the image names.

**Comment by mrkkrp** [comment]
Ping?

**Comment by bgamari** [comment]
Thanks for the ping, @mrkkrp!

I've granted you access to the `ghcci` Docker organization. I'm going to merge this today.

**Comment by mrkkrp** [comment]
Should I push the images there and amend the commit then?

**Comment by mrkkrp** [comment]
Pushed `ghcci/i386-linux:0.0.1` and amended the commit.

**Comment by mrkkrp** [comment]
Also pushed `ghcci/x86_64-linux:0.0.1`, we can switch that in a different PR perhaps?

**Comment by bgamari** [comment]
Merged with f9a6d4207fb0e551821fee847ac064ac31d96bba.

**Comment by bgamari** [comment]
Mark Karpov <notifications@github.com> writes:

> Also pushed `ghcci/x86_64-linux:0.0.1`, we can switch that in a different PR perhaps?
>
Yes, please do open a new PR changing the image locations.

**Pull request closed without merging**